NETDefault/Libs/Triple.KeyVault/KeyVaultConfigurationHelper.cs

```
@@ -0,0 +1,111 @@
+        /// </summary>
+        /// <param name="builder"></param>
+        public static void AddKeyVault(IFunctionsConfigurationBuilder builder)
+        {
+            var context = builder.GetContext();
+            AddKeyVault(builder.ConfigurationBuilder, context.ApplicationRootPath);
+        }
+#endif
+
+        private static void AddKeyVault(IConfigurationBuilder builder, string rootPath)
+        {
+            // ConfigurationBuilder already contains EnvironmentVariableProvider but we expect it to
+            // have the highest priority so we remove it and add it again after the appsettings.json provider
+            ConfigurationRemoveEnvironmentVariables(builder);
+
+            builder
+                .AddJsonFile(Path.Combine(rootPath, "appsettings.json"), optional: true, reloadOnChange: false)
+                .AddEnvironmentVariables();
+
+            var currentConfig = builder.Build();
+            var keyVaultName = currentConfig.GetValue<string>("KeyVaultName");
+            var keyVaultPrefix = currentConfig.GetValue<string>("KeyVaultPrefix");
+            var keyVaultRefreshInterval = currentConfig.GetValue<TimeSpan?>("KeyVaultInterval");
+
+            if (!string.IsNullOrWhiteSpace(keyVaultName))
+            {
+                try
+                {
+                    var secretClient = CreateSecretClient(keyVaultName);
+
+                    builder.AddAzureKeyVault(
+                        secretClient,
+                        new AzureKeyVaultConfigurationOptions
+                        {
+                            Manager = string.IsNullOrWhiteSpace(keyVaultPrefix)
+                                ? new KeyVaultSecretManager()
+                                : new PrefixKeyVaultSecretManager(keyVaultPrefix),
+                            ReloadInterval = keyVaultRefreshInterval
+                        });
+                }
+                catch
+                {
+                    return;
```

I know its just a demo project, but Isnt swallowing exceptions here setting a bad example?